tree/ntuple/v7/src/RNTuple.cxx

```diff
@@ -230,37 +230,26 @@ ROOT::Experimental::RNTupleReader *ROOT::Experimental::RNTupleReader::GetDisplay
    return fDisplayReader.get();
 }
 
-void ROOT::Experimental::RNTupleReader::Show(NTupleSize_t index, const ENTupleShowFormat format, std::ostream &output)
+void ROOT::Experimental::RNTupleReader::Show(NTupleSize_t index, std::ostream &output)
 {
-   RNTupleReader *reader = this;
-   REntry *entry = GetModel()->GetDefaultEntry();
-
-   switch (format) {
-   case ENTupleShowFormat::kCompleteJSON:
-      reader = GetDisplayReader();
-      entry = reader->GetModel()->GetDefaultEntry();
-      // Fall through
-   case ENTupleShowFormat::kCurrentModelJSON:
-      reader->LoadEntry(index);
-      output << "{";
-      for (auto iValue = entry->begin(); iValue != entry->end();) {
+   auto reader = GetDisplayReader();
```

Comment:
Now that `Show()` is inconditionally bound to the model of the reader, and as a small optiomization, I think we could also get rid of `fDisplayReader` and instead use the `void LoadEntry(NTupleSize_t index, REntry &entry)` overload in line 238 to avoid touching the default entry (?).